pkg/config/config.go

```diff
@@ -243,11 +241,10 @@ func (c *MicroshiftConfig) ReadFromCmdLine(flags *pflag.FlagSet) error {
 // Note: add a configFile parameter here because of unit test requiring custom
 // local directory
 func (c *MicroshiftConfig) ReadAndValidate(configFile string, flags *pflag.FlagSet) error {
-	if configFile == "" {
-		configFile = findConfigFile()
-	}
-	if err := c.ReadFromConfigFile(configFile); err != nil {
-		return err
+	if configFile != "" {
```

Comment:
It wasn't in the past -- that seems to have been an inadvertent change of the original PR.

Reply:
/hold

For this PR, are you proposing that it _should_ be a fatal error if neither a user nor a global config file exists? This has historically been treated as though there were an existent-but-empty config file, and only became an error in https://github.com/openshift/microshift/pull/1026#discussion_r1010559259. My first reaction was to restore the original behavior since there had been no discussion/intent to change it.

Reply:
I was proposing that this function consider an empty string an invalid filename. If we're saying the file doesn't have to exist (that makes sense), and the caller might pass an empty string when there is no file to read, then this function could treat the empty string as not an error.

Reply:
As it is, MicroShift follows your latter logic. The `configFile` is set by the `findConfigFile()` func (L143), which will return an empty string if no file is found.  The empty string value should continue to be interpreted as "use defaults" rather than throwing a fatal error.

Reply:
/unhold

The original behavior (i.e. to work without a config file) makes sense to me. I'm going to proceed with restoring it in this PR. If that behavior's not desirable, it can be changed deliberately sometime later.